### PR TITLE
feat: .io site nav CTA changes

### PR DESCRIPTION
**Pull request by EnMod**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-nq-cta-consistency-test-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1187888078520425/1202633580843430/f) 🎟️

## 🗒️ What / 🤷 Why

This PR reorders nav CTAs on our `_proxied-dot-io` subnavs, as part of a larger effort to make hero/nav CTAs more consistent across these pages. 

## 🧪 Testing

- [ ] Open the preview URL
- [ ] Check the following products' subnavs to ensure the primary CTA is "Try HCP [product]", by selecting a product in the bottom left of the page
  - [ ] Boundary
  - [ ] Vault
  - [ ] Packer
  - [ ] Consul

### CODE REVIEW

File: src/components/_proxied-dot-io/consul/subnav/index.jsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default function ConsulSubnav({ menuItems }) {
 				{ text: 'Download', url: '/downloads' },
 				{
 					text: 'Try HCP Consul',
-					url: 'https://cloud.hashicorp.com/?utm_source=consul_io&utm_content=top_nav_consul',
+					url: 'https://portal.cloud.hashicorp.com/sign-up?utm_source=docs&utm_content=top_nav_consul',
```
Comment (dstaley) [inline src/components/_proxied-dot-io/consul/subnav/index.jsx:24]: 🙋are these `utm_source` values correct? My first instinct is that they should have some consistency between them. `docs` seems weird given that we're using `vault_io` for Vault.

Comment (EnMod) [reply src/components/_proxied-dot-io/consul/subnav/index.jsx:24]: These are the desired values according to the stakeholder for this work.  These UTMs are in place just to ensure existing behavior on their end continues to work; they are intended to be removed at a later time.

**Review by alexcarpenter — APPROVED** [review]
LGTM

**Comment by EnMod** [comment]
@alexcarpenter I adjusted Packer's secondary CTA to read "Download" instead of "Install Packer", for consistency.  Going to merge this shortly!

**Pull request merged**